hayeon/programmers-2/3차-압축.js

```diff
@@ -0,0 +1,32 @@
+function solution(msg) {
+  const dictionary = Array(26)
+    .fill(0)
+    .map((_, idx) => String.fromCharCode(idx + 65));
+  const answer = [];
+
+  const compression = msg.split("").reduce((acc, cur) => {
+    const nextWord = acc + cur;
+    const nextWordIndex = dictionary.indexOf(nextWord);
+    const prevWordIndex = dictionary.indexOf(acc);
+
+    if (nextWordIndex === -1) {
+      answer.push(prevWordIndex + 1);
+      dictionary.push(nextWord);
+      return cur;
+    } else {
+      return nextWord;
+    }
+  }, "");
+
+  answer.push(dictionary.indexOf(compression) + 1);
+  return answer;
+}
```

Comment:
```suggestion
function solution(msg) {
  const dictionary = Array.from({ length: 26}, (_, i) => String.fromCharCode(idx + 65));
  const answer = [];

  const compression = msg.split("").reduce((acc, cur) => {
    const nextWord = acc + cur;
    const nextWordIndex = dictionary.indexOf(nextWord);
    const prevWordIndex = dictionary.indexOf(acc);

    if (nextWordIndex !== -1) return nextWord;
    
    dict.push(nextWord);
    
    if (prevWordIndex !== -1) answer.push(prevWordIndex + 1);
    return answer;
  }, "");

  answer.push(dictionary.indexOf(compression) + 1);
  return answer;
}
```
- 풀이잘봤습니다
- Array쪽은 fill, map을 Array.from으로 한번에 처리해줄 수 있습니다!
- reduce 내부에서 return문을 활용하면 좀 더 블록문을 제거해서 가독성을 키울 수 있습니다
- 해당 로직에대한 이해는 잘되셨을까요 😂 고생하셨습니다